### ISSUE TITLE
Mail driver problems

### ISSUE THREAD

**Issue by major-mayer** [Mateus-Romera/docker-snipe-it#4]
Hi @Mateus-Romera ,
thanks for the quick fix of the last problem regard file access.
That's not happening anymore with the new version, but i still got some problems that prevent the container from starting.
In particular this:
```
  Please use the argument -v to see more details.
   InvalidArgumentException  : Driver [${MAIL_DRIVER:-smtp}] not supported.
  at /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Manager.php:119
    115|                 return $this->$method();
    116|             }
    117|         }

  > 119|         throw new InvalidArgumentException("Driver [$driver] not supported.");

    120|     }
122|     /**

    123|      * Call a custom driver creator.
  Exception trace:
  1   Illuminate\Support\Manager::createDriver("${MAIL_DRIVER:-smtp}")
      /var/www/html/vendor/laravel/framework/src/Illuminate/Support/Manager.php:90

  2   Illuminate\Support\Manager::driver()
      /var/www/html/vendor/laravel/framework/src/Illuminate/Mail/MailServiceProvider.php:94

  Please use the argument -v to see more details.
```

Do i need to set some mail related environment variables before i can start the container?

**Comment by Mateus-Romera** [comment]
Hello @major-mayer ,
Sorry, you should not have this much errors using a new image :/

Are you sure you are using the latest image? I did not change the tag so you might need to force pull the image from docker hub. Could you try doing that?
This issue should already be resolved.

You only need the environment variables described in this example: https://github.com/Mateus-Romera/docker-snipe-it/blob/main/examples/simple/docker-compose.yaml

**Comment by major-mayer** [comment]
Hi sorry for my late reply.
I just tried pulling the latest image, but it seems already up to date:
```bash
laurenz@laurenz-BHYVE:~$ docker pull  pirasbro/snipe-it:5.3.0-fpm-alpine
5.3.0-fpm-alpine: Pulling from pirasbro/snipe-it
Digest: sha256:0f43fa49c27c79c23365ee5c309447601295a54c83ea94c59bcf99b7400ef45c
Status: Image is up to date for pirasbro/snipe-it:5.3.0-fpm-alpine
docker.io/pirasbro/snipe-it:5.3.0-fpm-alpine

```

But still the error persists.
Do you see any configuration errors in my docker-compose file?

```yaml
version: '3'

volumes:
  web_data:
  snipe_data : 
  
services:
  snipeit:
    container_name: snipeit-fpm
    image: pirasbro/snipe-it:5.3.0-fpm-alpine
    environment:
      # SnipeIT settings
      APP_URL: localhost:9001
      APP_KEY: base64:CdNywz90QeGfAvL4nw7sqHwr9nI8v4z+yYcWpEgH+4g=
      # Database settings
      DB_HOST: mariadb
      DB_PORT: 3306
      DB_DATABASE: snipeit
      DB_USERNAME: snipeit
      DB_PASSWORD: xxx
      # Session settings
      SECURE_COOKIES: "false"
    volumes:
      - web_data:/var/www/html
      - snipe_data:/var/lib/snipeit
``

**Comment by major-mayer** [comment]
All right, i had to clean up the existing volumes and reconfigure the web-server and database, but now it works for me.
I think the image is fine :slightly_smiling_face:  Thank you very much for your work.
Do you plan to keep the image up to date?

**Comment by Mateus-Romera** [comment]
@major-mayer Hello again, hope you are doing well!

I'm happy the image is finally working for you!

And yes, I'm definitely keeping the image up to date. Probably doing this tonight or tomorrow.

**Comment by major-mayer** [comment]
Great to hear that ❤️

**Comment by Mateus-Romera** [comment]
Hello again, @major-mayer 

I just updated the image.
Let me know If you encounter any problems!
Have a great day!

**Comment by major-mayer** [comment]
Hi,
I just recreated the container with the new image, and everything seems to be working fine.
On thing i noticed tho:

In the settings it says that Snipe-It is now on :  v5.3.0 build 6470 (gf87ffb84d) 
while the environment variable of the image says: SNIPEIT_RELEASE=5.3.3

I'm not exactly sure which value i should trust now and which is correct.

**Comment by Mateus-Romera** [comment]
This is probably due to the volume you are using on your `docker-compose.yml` between snipeit and nginx services. It is persisting the old installation files. I should remove this volume from my example docker-compose.yml too.

You could remove this:
```
volumes:
  - web_data:/var/www/html
```

I just can't test all the implications for this action right now.

**Comment by Mateus-Romera** [comment]
Actually, forgot what I said earlier. You just need to delete the `web_data` volume for every new snipeit release, otherwise it will persist the old installation files in the snipeit and nginx services. Right now I don't have a better solution, but I will look into this problem later on.

**Comment by major-mayer** [comment]
Ah interesting...actually some time ago I also created my own php fpm version of snipe and was struggling with exactly the same update problems.
Good to see that I am not alone 😅

I found this thread https://stackoverflow.com/questions/61668433/serving-new-static-files-via-nginx-stored-in-php-fpm-container-after-update
Maybe the third point of the second answer could work for this case?
Copying all files to a different location and restoring them afterwards seems doable.

It would be interesting to see what nextcloud e.g. are doing in their fpm image
